### PR TITLE
feat(secrets): Keep original secrets data in runtime for further validation

### PR DESCRIPTION
**Pull request by marynaKK**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

added a class to save secrets data for further validation.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/secrets/coordinator.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class SecretsCoordinator:
+    def __init__(self) -> None:
+        self._secrets: Dict[str, EnrichedSecret] = {}
+
+    def add_secret(self, enriched_secret: EnrichedSecret, check_result: _CheckResult) -> None:
+        # can be changed to any other suitable way.
+        # should not have duplicates? - if duplicates allowed, implementation should be changed
+        # may be saved by file type first, then by key - or any other preprocessing that may help differ the secrets.
+
+        if check_result.get('result') == CheckResult.FAILED and enriched_secret.original_secret is not None:
```
Comment (Eliran-Turgeman) [inline checkov/secrets/coordinator.py:25]: We can check these conditions before creating the `EnrichedSecret` object, in the runner itself

File: checkov/secrets/coordinator.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from typing import Iterable, Dict, Optional
+
+
+class EnrichedSecret:
+    __slots__ = ("original_secret", "bc_check_id", "resource")
```
Comment (gruebel) [inline checkov/secrets/coordinator.py:5]: nice 🍪

File: checkov/secrets/coordinator.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    def __init__(self, original_secret: Optional[str], bc_check_id: str, resource: str):
```
Comment (gruebel) [inline checkov/secrets/coordinator.py:7]:
```suggestion
    def __init__(self, original_secret: Optional[str], bc_check_id: str, resource: str) -> None:
```

File: checkov/secrets/coordinator.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        self.original_secret = original_secret
+        self.bc_check_id = bc_check_id
+        self.resource = resource
+
+
+class SecretsCoordinator:
```
Comment (gruebel) [inline checkov/secrets/coordinator.py:13]: maybe also using here `__slots__`

**Review by lirshindalman — APPROVED** [review]

**Review by Eliran-Turgeman — APPROVED** [review]
🔥

**Pull request merged**